planner/cascades/transformation_rules.go

```diff
@@ -323,3 +326,88 @@ func (r *PushSelDownProjection) OnTransform(old *memo.ExprIter) (newExprs []*mem
 	newTopSelExpr.SetChildren(newProjGroup)
 	return []*memo.GroupExpr{newTopSelExpr}, true, false, nil
 }
+
+// PushSelDownAggregation pushes Selection down to the child of Aggregation.
+type PushSelDownAggregation struct {
+}
+
+// GetPattern implements Transformation interface.
+// The pattern of this rule is `Selection -> Aggregation`.
+func (r *PushSelDownAggregation) GetPattern() *memo.Pattern {
+	return memo.BuildPattern(
+		memo.OperandSelection,
+		memo.EngineTiDBOnly,
```

Comment:
Why it's only for TiDB?

Reply:
`EngineAll` would be better for now, fixed.